lib/kube/proxy/forwarder.go

```diff
@@ -1718,7 +1718,7 @@ func setupImpersonationHeaders(log logrus.FieldLogger, ctx authContext, headers
 
 	// Make sure to overwrite the exiting headers, instead of appending to
 	// them.
-	headers[ImpersonateGroupHeader] = nil
+	headers.Del(ImpersonateGroupHeader)
```

Comment:
Does it make sense to test for this scenario, so we avoid regressions?

Reply:
Done in 983117bfa430da1a2d5add537d90766d62683121